openhexa/sdk/pipelines/parameter.py

```diff
@@ -487,7 +487,12 @@ def _validate_default(self, default: typing.Any, multiple: bool):
         except ParameterValueError:
             raise InvalidParameterError(f"The default value for {self.code} is not valid.")
 
-        if self.choices is not None and default not in self.choices:
+        if self.choices is not None:
+            if isinstance(default, list):
+                if not all(d in self.choices for d in default):
+                    raise InvalidParameterError(
+                        f"The default list of values for {self.code} is not included in the provided choices."
+                    )
             raise InvalidParameterError(f"The default value for {self.code} is not included in the provided choices.")
```

Comment:
So if `self.choices is not None` it's going to raise an `InvalidParameterError` in any case ? What if `default` is not a list because `multiple=False`?

Reply:
Oh yes, I should not remove the initial check.